src/UserCredentials.cs

```diff
@@ -186,6 +217,38 @@ private static void ValidatePassword(SecureString password)
             if (password.Length == 0)
                 throw new ArgumentException("Password cannot be empty.", nameof(password));
         }
+        
+		private static void ValidateUserWithoutDomainForEmptyPassword(string username)
+		{
+			if (username.EndsWith(@"$"))
+				return;
+
+			var specialAccounts = new[]
+			{
+				new SecurityIdentifier(WellKnownSidType.LocalServiceSid, null),
+				new SecurityIdentifier(WellKnownSidType.LocalSystemSid, null),
+				new SecurityIdentifier(WellKnownSidType.NetworkServiceSid, null),
+			};
```

Comment:
Since `specialAccounts` is fixed and used twice, please make it a private static readonly field on the class.